src/Terrain/Environment.java

```diff
@@ -20,6 +21,42 @@ private Environment(Terrains tileType, Weathers weather)
 	{
 		terrainType = tileType;
 		weatherType = weather;
+		// If there's a better way to do this, I'm all ears
+		switch(terrainType){
+		case PLAIN:
+			defLevel = 1;
+			break;
+		case FOREST:
+			defLevel = 2;
+			break;
+		case MOUNTAIN:
+			defLevel = 4;
+			break;
+		case ROAD:
+			defLevel = 0;
+			break;
+		case CITY:
+			defLevel = 3;
+			break;
+		case FACTORY:
+			defLevel = 4;
+			break;
+		case HQ:
+			defLevel = 5;
+			break;
+		case SHOAL:
+			defLevel = 0;
+			break;
+		case WATER:
+			defLevel = 1;
+			break;
+		case REEF:
+			defLevel = 2;
+			break;
+		default:
+			defLevel = -1;
```

Comment:
We should at least print an error here rather than just continue happily along with bad data.